src/data-connect/data-connect-api.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * @license
+ * Copyright 2024 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ConnectorConfig {
+	location: string;
+	serviceId: string;
+}
+
+export interface ExecuteGraphqlResponse<GraphqlResponse> {
+  data: GraphqlResponse;
+}
+
+export interface GraphqlOptions<Variables> {
```

Comment:
Are there any restrictions at all about what Variables and GraphqlResponse (above) can be? Should Variables always be a map/object? If so maybe a little more restriction like:
```
export interface GraphqlOptions<Variables extends Record<string, unknown>>
```
to indicate Variables must be a map/object (or in TS, Record)?

I think GraphqlResponse also seems like it is supposed to be a map/object?

Reply:
Good call. Yes, Variables is always a map/object. I will update this to make it more restrictive

Reply:
Discussed offline. Making the types more restrictive throws for plain interface types, unless they are contain index signatures. Leaving it as is for now.